lib/Test/PostgreSQL.pm

```diff
@@ -399,12 +399,14 @@ method _find_port_and_launch() {
     catch {
       # warn "Postgres failed to start on port $port\n";
       unless ($tries--) {
-        die "Failed to start postgres on port $port: $_";
+        die "Failed to start postgres after trying 10 potential ports: $_";
       }
       undef;
     };
     return if $good;
-    $port++;
+    # Increment port by a random number to avoid clashes with other Test::Postgresql processes
+    # Keep in mind that this increment is going to be made up to 10 times, so avoid exceeding 64k
+    $port += int(rand(500)) + 1;
```

Comment:
I'm not a Perlmonger by any means but: shouldn't we seed with `srand()` first?

Reply:
I haven't written any Perl for several years, so you're probably more of a Perlmonger than I am, at this point.

However I believe you don't need to call srand(), at least not on any modern build of Perl.

Reply:
So, I did a bit more testing and...

By calling `srand()` just before the `while` loop begins, I have yet to see colliding ports. After commenting out the `srand()` call, I immediately see a colliding port.

I don't know why this works... but it works, and it's a simple fix:

```diff
diff --git a/lib/Test/PostgreSQL.pm b/lib/Test/PostgreSQL.pm
index 55728f5..eb75938 100644
--- a/lib/Test/PostgreSQL.pm
+++ b/lib/Test/PostgreSQL.pm
@@ -390,6 +390,7 @@ method start() {
 method _find_port_and_launch() {
   my $tries = 10;
   my $port = $self->base_port;
+  srand();
   # try by incrementing port number until PostgreSQL starts
   while (1) {
     my $good = try {
```

Reply:
That's interesting, and confusing.
I'm happy to include that if it's a fix. I'm also curious to understand how you're using the library.

Is your code loading Test::PostgreSQL, and then using fork() to split into many processes?
I think that's a known scenario where Perl's RNG needs reseeding.